apps/api/src/applicant/external-api.service.ts

```diff
@@ -650,14 +649,25 @@ export class ExternalAPIService {
     }
   }
 
-  async getApplicants(start?: string, end?: string) {
+  async getApplicants(filter: IENUserFilterAPIDTO) {
+    const { from, to, limit, skip } = filter;
+    const queryString = `SELECT DISTINCT applicant_id  
+    FROM ien_applicant_status_audit 
+    WHERE updated_date <'${to || dayjs().add(1, 'day').format('YYYY-MM-DD')}' and updated_date >'${
+      from || '1914-07-18'
+    }';`;
+    const ids: string[] = (await this.ienapplicantStatusAuditRepository.query(queryString)).map(
+      (result: { applicant_id: string }) => result.applicant_id,
+    );
+    if (!ids.length) {
+      return [];
+    }
     return this.ienapplicantRepository.find({
-      where: {
-        updated_date: Between(
-          new Date(start || '1918-07-18').toISOString(),
-          end ? new Date(end).toISOString() : new Date().toISOString(),
-        ),
+      where: (qb: any) => {
+        qb.where(`IENApplicant.id IN (:...ids)`, { ids });
```

Comment:
How about using the sub list of ids instead of `skip` and `take`?

Reply:
Thats a great idea, it should speed up the query.